### PR TITLE
Update silo data view

### PR DESCRIPTION
**Pull request by jefmoura**
## Purpose
The data view of silo should support query and aggregation.

## Approach
Added a new query parameter to receive aggregation action and perform them to the data.

### Furter Info
Related ticket: #444

### CODE REVIEW

File: silo/api.py [hunk 1]
```diff
@@ -302,28 +300,59 @@ def data(self, request, id):
...
+            data = JSONEncoder().encode(result)
         else:
-            data = LabelValueStore.objects(silo_id=id, **filter_fields).exclude('create_date', 'edit_date', 'silo_id','read_id')
-
-        sort = str(request.GET.get('sort',''))
-        data = data.order_by(sort)
-        json_data = json.loads(data.to_json())
-
-        return JsonResponse({"data": json_data, "draw": draw, "recordsTotal": recordsTotal, "recordsFiltered": recordsTotal}, safe=False)
+            cmd_cur = collection.aggregate(pipeline=pipeline)
```
Comment (Menda) [inline silo/api.py]: This should not be there. This operation was done before.

File: silo/api.py [hunk 2]
```diff
@@ -302,28 +300,59 @@ def data(self, request, id):
...
+            result = list(cmd_cur)
```
Comment (Menda) [inline silo/api.py]: Delete this and better do in line 335:

```
result = list(cmd_cur)
records_total = result[0]['count']
```

File: silo/api.py [hunk 3]
```diff
@@ -302,28 +300,59 @@ def data(self, request, id):
...
+            pipeline.append({'$sort': sort_fields})
+
+        count_pipeline = pipeline + [{'$count': 'count'}]
+        cmd_cur = collection.aggregate(pipeline=count_pipeline)
+        result = list(cmd_cur)[0]
+        records_total = result['count']
+
+        # workaround until the problem of javascript not increasing
+        # the value of length is fixed
         if offset >= 0:
```
Comment (Menda) [inline silo/api.py]: If you know the `offset` before, finish appending to the pipeline before you start doing operations with the database

File: silo/api.py [hunk 4]
```diff
@@ -302,28 +300,59 @@ def data(self, request, id):
...
-            data = LabelValueStore.objects(silo_id=id, **filter_fields).exclude('create_date', 'edit_date', 'silo_id','read_id')
-
-        sort = str(request.GET.get('sort',''))
-        data = data.order_by(sort)
-        json_data = json.loads(data.to_json())
-
-        return JsonResponse({"data": json_data, "draw": draw, "recordsTotal": recordsTotal, "recordsFiltered": recordsTotal}, safe=False)
+            cmd_cur = collection.aggregate(pipeline=pipeline)
+            result = list(cmd_cur)
+            data = JSONEncoder().encode(result)
```
Comment (Menda) [inline silo/api.py]: This operation is repeated inside the `if`

Comment (jefmoura) [reply silo/api.py]: - This `offset` is only used for the calculation of the length the if.
- `offset` action in MongoDB doesn't accept negative values as well

File: silo/api.py (hunk 1 above)
Comment (jefmoura) [reply silo/api.py]: The first operation only counts the number of rows. This one we return the data.

File: silo/api.py (hunk 3 above)
Comment (jefmoura) [reply silo/api.py]: - This `offset` is only used for the calculation of the length the if.
- `offset` action in MongoDB doesn't accept negative values as well

File: silo/api.py (hunk 4 above)
Comment (jefmoura) [reply silo/api.py]: True. I can move it out from the conditional.

File: silo/api.py (hunk 2 above)
Comment (jefmoura) [reply silo/api.py]: This `result` doesn't have `count`.

**Pull request merged**